packages/txservice/src/error.ts

```diff
@@ -220,35 +220,104 @@ export class TransactionKilled extends TransactionError {
   }
 }
 
-export class TransactionServiceFailure extends NxtpError {
-  /**
-   * An error that indicates that transaction service infrastructure had a critical
-   * and unexpected failure.
-   */
-  static readonly type = TransactionServiceFailure.name;
+export class MaxAttemptsReached extends NxtpError {
+  static readonly type = MaxAttemptsReached.name;
+
+  static getMessage(attempts: number): string {
+    return `Reached maximum attempts ${attempts}.`;
+  }
+
+  constructor(attempts: number, public readonly context: any = {}) {
+    super(MaxAttemptsReached.getMessage(attempts), context, MaxAttemptsReached.type);
+  }
+}
+
+export class NotEnoughConfirmations extends NxtpError {
+  static readonly type = NotEnoughConfirmations.name;
+
+  static getMessage(required: number, hash: string, confs: number): string {
+    return `Never reached the required amount of confirmations (${required}) on ${hash} (got: ${confs}). Did a reorg occur?`;
+  }
+
+  constructor(required: number, hash: string, confs: number, public readonly context: any = {}) {
+    super(NotEnoughConfirmations.getMessage(required, hash, confs), context, NotEnoughConfirmations.type);
+  }
+}
+
+export class GasEstimateInvalid extends NxtpError {
+  static readonly type = GasEstimateInvalid.name;
+
+  static getMessage(returned: string): string {
+    return `The gas estimate returned was an invalid value. Got: ${returned}`;
+  }
+
+  constructor(returned: string, public readonly context: any = {}) {
+    super(GasEstimateInvalid.getMessage(returned), context, GasEstimateInvalid.type);
+  }
+}
+
+export class ProviderNotConfigured extends NxtpError {
+  static readonly type = ProviderNotConfigured.name;
+
+  static getMessage(chainId: string): string {
+    return `No provider(s) configured for chain ${chainId}. Make sure this chain's providers are configured.`;
+  }
+
+  constructor(public readonly chainId: string, public readonly context: any = {}) {
+    super(ProviderNotConfigured.getMessage(chainId), context, ProviderNotConfigured.type);
+  }
+}
+
+export class ConfigurationError extends NxtpError {
+  static readonly type = ConfigurationError.name;
+
+  constructor(public readonly invalidParamaters: any, public readonly context: any = {}) {
+    super("Configuration paramater(s) were invalid.", { ...context, invalidParamaters }, ConfigurationError.type);
+  }
+}
+
+export class InitialSubmitFailure extends NxtpError {
+  static readonly type = InitialSubmitFailure.name;
+
+  constructor(public readonly context: any = {}) {
+    super(
+      "Transaction never submitted: exceeded maximum iterations in initial submit loop.",
+      context,
+      InitialSubmitFailure.type,
+    );
+  }
+}
+
+// These errors should essentially never happen; they are only used within the block of sanity checks.
```

Comment:
willing to let it slide then

Reply:
yeah. i think this was the original intention with TransactionServiceFailure: make a grouping of errors used in sanity checks only that we could look out for to indicate extremely unexpected misbehavior. Realistically, these checks exist just to *eliminate* the possibility of a future problem being one of those cases

This error "grouping" feels a lot more specific to that imo